app/services/booking_bug.rb

```diff
@@ -17,4 +17,22 @@ def self.config
     yield(@config) if block_given?
     @config
   end
+
+  # Allow subset of actions to be run by specifying the number of actions to be performed.
+  # Defaults to all actions
+  def call(actions_to_perform: actions.count)
+    actions[0..actions_to_perform - 1].inject([]) do |data, action|
+      action.call(data)
+    end
+  end
+
+  def actions
+    @actions ||= [
+      Etl::Api.new(
+        base_path: "/api/v1/admin/#{BookingBug.config.company_id}/bookings",
+        connection: BookingBugConnection.new(config: BookingBug.config)
+      ),
+      Etl::Filter.new { |record| Facts::Booking.where(reference_number: record['id']).empty? }
+    ]
+  end
 end
```

Comment:
This implementation is still completely subject to change. 

I have gone with an array of processor as that feels like it is closest to a pipeline.

Thought/alternatives welcome..